modules/libcom/test/iocshTest.cpp

```diff
@@ -83,14 +84,85 @@ void assertCallFunc(const iocshArgBuf *args)
     iocshSetError(args[0].ival);
 }
 
+const iocshFuncDef testHelpFunction1Def = {"testHelpFunction1",0,0,
+                                    "Usage message of testHelpFunction1\n"};
+const iocshFuncDef testHelpFunction2Def = {"testHelpFunction2",0,0,
+                                    "Usage message of testHelpFunction2\n"};
+const iocshFuncDef testHelpFunction3Def = {"testHelpFunction3",0,0,
+                                    "Usage message of testHelpFunction3\n"};
+void doNothing(const iocshArgBuf *args)
+{
+    return;
+}
+
+std::string readFile(std::string filename)
+{
+    std::ifstream t(filename.c_str());
```

Comment:
Should check for `fail()` or `!is_open()` and throw an exception for a cleaner error if a test file is missing?

Reply:
Added in 561235d987dda3f42bf2481f2fd899e83b38ae24.

I note that raising the exception causes all subsequent tests to be skipped - I assume this what you intended?

Reply:
I haven't checked the code, is that exception just skipping the tests which depend on that file, or any/the tests beyond that as well? The correct behaviour should be that being unable to open a file will fail a test for that file opening, and subsequent tests that depend on that file should be skipped and something like `testSkip("File open failed", 5);` called instead (where 5 is the number of tests that can't be run).

BTW you can ignore the test failures from Appveyor in unrelated test programs, we are aware that some of our tests do fail there. Again, my apologies that we haven't finished reviewing and merging all of the Codeathon PRs yet.

Reply:
> I note that raising the exception causes all subsequent tests to be skipped - I assume this is what you intended?

What I would like to avoid is having the following tests using this input silently given an empty string.  Whether this lack of silence is an exception, returning (and acting on) an error code, or just a `testFail()`, is less important to me.